sql/plan/show_create_table.go

```diff
@@ -255,7 +266,16 @@ func (i *showCreateTablesIter) produceCreateTableStatement(ctx *sql.Context, tab
 
 		// TODO: The columns that are rendered in defaults should be backticked
 		if col.Default != nil {
-			stmt = fmt.Sprintf("%s DEFAULT %s", stmt, col.Default.String())
+			defStr := col.Default.String()
+			if defStr == "NULL" || !col.Default.IsLiteral() || sql.IsTime(col.Default.Type()) || sql.IsText(col.Default.Type()) {
```

Comment:
(minor) I would flip the logic on this if statement so you don't have an empty if block with all the code in the else block.